mongo/integration/crud_prose_test.go

```diff
@@ -113,6 +114,136 @@ func TestWriteErrorsWithLabels(t *testing.T) {
 
 }
 
+func TestWriteErrorsDetails(t *testing.T) {
+	clientOpts := options.Client().
+		SetRetryWrites(false).
+		SetWriteConcern(mtest.MajorityWc).
+		SetReadConcern(mtest.MajorityRc)
+	mtOpts := mtest.NewOptions().
+		ClientOptions(clientOpts).
+		MinServerVersion("5.0").
+		Topologies(mtest.ReplicaSet, mtest.Single).
+		CreateClient(false)
+
+	mt := mtest.New(t, mtOpts)
+	defer mt.Close()
+
+	mt.Run("JSON Schema validation", func(mt *mtest.T) {
+		// Create a JSON Schema validator that requires properties "a" and "b".
+		jsonSchema := bson.M{
+			"$jsonSchema": bson.M{
+				"bsonType": "object",
+				"required": []string{"a", "b"},
+				"properties": bson.M{
+					"a": bson.M{
+						"bsonType": "string",
+					},
+					"b": bson.M{
+						"bsonType": "int",
+					},
+				},
+			},
+		}
+
+		// Create a collection with a random name and set a JSON Schema validator on it.
+		collName := mtest.RandomString(10)
+		err := mt.DB.CreateCollection(
```

Comment:
I know we talked about this a bit in slack, but it's still unclear to me why all the tests have to run on the exact same collection instead of letting the subtests each create their own collection with the collection options set.

Reply:
Re-evaluating this after some investigation, I think using the `CollectionCreateOptions` is probably the better solution. There's no reason why all the subtests need to use the same collection, it was just a simplification compared to having all the subtests create their own collection if we have to create that collection in the subtest code.

Will update.